overlord/assertstate/assertstate.go

```diff
@@ -1129,3 +1129,87 @@ func TemporaryDB(st *state.State) *asserts.Database {
 	db := cachedDB(st)
 	return db.WithStackedBackstore(asserts.NewMemoryBackstore())
 }
+
+// ValidationSetsModelFlags contains flags for ValidationSetsFromModel.
+type ValidationSetsModelFlags struct {
+	// Offline should be set to true if the store should not be accessed. Any
+	// assertions will be retrieved from the existing assertions database. If
+	// the assertions are not present in the database, an error will be
+	// returned.
+	Offline bool
+}
+
+// ValidationSetsFromModel takes in a model and creates a
+// snapasserts.ValidationSets from any validation sets that the model includes.
+func ValidationSetsFromModel(st *state.State, model *asserts.Model, store snapstate.StoreService, flags ValidationSetsModelFlags) (*snapasserts.ValidationSets, error) {
```

Comment:
could this take a DeviceContext that can give a store instead of a store directly? this would be the only function here that takes a StoreService

Reply:
Thanks, that makes sense. Made that change.